File: SoftwareEngineering/UnitTestsExample/Java Example/HypotenuseTest.java
import org.junit.Test;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;

public class HypotenuseTest extends Hypotenuse {

	@Test
	public void testCalculateHypotenuse(){
		// 3^2 + 4^2 = 5^2
		assertEquals(5.0, Hypotenuse.calculateHypotenuse(3, 4), 0.01);
	}

	@Test
    public void testAnotherCorrectCase(){
        // 6^2 + 8^2 = 10^2
        assertEquals(10.0, Hypotenuse.calculateHypotenuse(6, 8), 0.01);
    }

    @Test
    public void testAnIncorrectCase(){
        // 9^2 + 16^2 != 30, rather something like 18.357-ish
        assertNotEquals(30.0, Hypotenuse.calculateHypotenuse(9, 16), 0.01);
    }
}

File: SoftwareEngineering/UnitTestsExample/Python Example/HypotenuseTest.py
import unittest

class TestCases(unittest.TestCase):

	# First Test
	def testCalculateHypotenuse():
		assert 5 == calculate_hypotenuse(3, 4)

	# Second Test
	def testAnotherCorrectCase():
		assert 10 == calculate_hypotenuse(6, 8)

	# Negative logic testing 
	def testAnIncorrectCase():
		assert 30 != calculate_hypotenuse(9, 16)

# Runs all of the tests
if __name__ == "__init__":
    unittest.main() # This will call the tests above